### PR TITLE
docs: Fix deb download link

### PR DESCRIPTION
**Pull request by jszwedko**
Closes: #12243

I will symlink all existing deb artifacts in S3 to their new names.

Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

File: website/cue/reference/administration/downloads.cue [hunk 1]
```diff
@@ -24,7 +24,12 @@ administration: {
 			}
 
 			// Calculate the download URL without needing site templating
-			download_url: "\(urls.vector_packages_root)/vector/{v1}/vector-{v2}-\(_version_postfix)\(filename)"
+			if _file_type != "deb" {
+				download_url: "\(urls.vector_packages_root)/vector/{v1}/vector-{v2}-\(_version_postfix)\(filename)"
+			}
+			if _file_type == "deb" {
+				download_url: "\(urls.vector_packages_root)/vector/{v1}/vector_{v2}-1_\(_version_postfix)\(filename)"
+			}
```
Comment (jszwedko) [inline website/cue/reference/administration/downloads.cue:32]: This is kinda hacky, but I think I'm ok with it until we have more variations.

File: website/cue/reference/administration/downloads.cue [hunk 2]
```diff
@@ -24,7 +24,12 @@ administration: {
 			}
 
 			// Calculate the download URL without needing site templating
-			download_url: "\(urls.vector_packages_root)/vector/{v1}/vector-{v2}-\(_version_postfix)\(filename)"
+			if _file_type != "deb" {
+				download_url: "\(urls.vector_packages_root)/vector/{v1}/vector-{v2}-\(_version_postfix)\(filename)"
```
Comment (spencergilbert) [inline website/cue/reference/administration/downloads.cue:28]: I read that as "version post fix" 😆

**Review by spencergilbert — APPROVED** [review]
As long as we symlink before merging, lgtm

**Comment by jszwedko** [comment]
Symlinks created. I'll cherry-pick this to the release branch.

For posterity, to create the symlinks I:

* `aws s3api list-objects-v2 --bucket packages.timber.io --prefix vector > /tmp/objects.json`
* `cat /tmp/objects.json| jq '.Contents[] | select(.Key|test("vector-(0.[^-]+|nightly)-[^-]+.deb$")) | .Key' -r > /tmp/objects.txt`
* `for key in $(cat /tmp/objects.txt); do echo aws s3api put-object --bucket packages.timber.io --key "$(echo $key | sed -E 's/vector-(.*)-(.*).deb/vector_\1-1_\2.deb/')" --website-redirect-location "/$key" ; done > /tmp/to-run.txt`
* Verified commands
* `cat /tmp/to-run.txt |  parallel --eta`

**Pull request merged**